src/jit/morph.cpp

```diff
@@ -4389,18 +4592,65 @@ GenTreePtr    Compiler::fgMorphMultiregStructArg(GenTreePtr arg, fgArgTabEntryPt
         assert(varNum < lvaCount);
         LclVarDsc* varDsc = &lvaTable[varNum];
 
-        // At this point any TYP_STRUCT LclVar must be a 16-byte pass by value argument
-        assert(varDsc->lvSize() == 2 * TARGET_POINTER_SIZE);
+        // At this point any TYP_STRUCT LclVar must be a 16-byte struct
+        // or an HFA struct, both which are passed by value.
+        //
+        assert((varDsc->lvSize() == 2*TARGET_POINTER_SIZE) || varDsc->lvIsHfa());
+
+        varDsc->lvIsMultiRegArgOrRet = true;
 
-        const BYTE * gcPtrs = varDsc->lvGcLayout;
+#ifdef DEBUG
+        if (verbose)
+        {
+            JITDUMP("Multireg struct argument V%02u : ");
+            fgEntryPtr->Dump();
+        }
+#endif // DEBUG
 
-        var_types type0 = getJitGCType(gcPtrs[0]);
-        var_types type1 = getJitGCType(gcPtrs[1]);
+        // This local variable must match the layout of the 'objClass' type exactly
+#ifdef FEATURE_HFA
+        if (varDsc->lvIsHfa())
+        {
+            // We have a HFA struct
+            noway_assert(elemType == varDsc->lvHfaTypeIsFloat ? TYP_FLOAT : TYP_DOUBLE);
```

Comment:
There are only very few occurrences of reference to lvHJfaTypeIsFloat.  Consider adding an accessor method for it that it return false for non-hfa targets.  that would allow the above #ifdef to be removed.